web/src/api/system/software.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) [2024-2025] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+type System = {
+  addons: AddonInfo[];
+  patterns: Pattern[];
+  repositories: Repository[];
+};
+
+type Pattern = {
+  /** Pattern name (internal ID) */
+  name: string;
+  /** Pattern category */
+  category: string;
+  /** User visible pattern name */
+  summary: string;
+  /** Long description of the pattern */
+  description: string;
+  /** {number} order - Display order (string!) */
```

Comment:
I am really confused here what means that `string!` part in comment.

Reply:
I only moved the existing code. This will be revisited.